Aplib.Core/Desire/Goal.cs

```diff
@@ -0,0 +1,79 @@
+namespace Aplib.Core.Desire
+{
+    public class Goal
+    {
+        /// <summary>
+        /// The goal is considered to be completed, when the distance of the
+        /// <see cref="CurrentHeuristics"/> is below this value.
+        /// </summary>
+        private const float Epsilon = 0.005f;
+
+        /// <summary>
+        /// The <see cref="Heuristics"/> of the current state of the game.
+        /// </summary>
+        public Heuristics CurrentHeuristics { get; private set; }
+
+        /// <summary>
+        /// The <see cref="Tactic"/> used to achieve this <see cref="Goal"/>, which is executed during every iteration
+        /// of the BDI cycle.
+        /// </summary>
+        /// <seealso cref="Iterate()"/>
+        private readonly Tactic _tactic;
+        /// <summary>
+        /// The <see cref="IGoalPredicate"/> used to test whether this <see cref="Goal"/> has been completed.
+        /// </summary>
+        /// <seealso cref="IsCompleted()"/>
+        private readonly IGoalPredicate _goalPredicate;
```

Comment:
Nitpick: missing enter here:
```suggestion
        /// <summary>
        /// The <see cref="Tactic"/> used to achieve this <see cref="Goal"/>, which is executed during every iteration
        /// of the BDI cycle.
        /// </summary>
        /// <seealso cref="Iterate()"/>
        private readonly Tactic _tactic;

        /// <summary>
        /// The <see cref="IGoalPredicate"/> used to test whether this <see cref="Goal"/> has been completed.
        /// </summary>
        /// <seealso cref="IsCompleted()"/>
        private readonly IGoalPredicate _goalPredicate;
```